modules/core/exported/channel.go

```diff
@@ -34,6 +34,10 @@ type PacketI interface {
 // Acknowledgement defines the interface used to return
 // acknowledgements in the OnRecvPacket callback.
 type Acknowledgement interface {
+	// Success tells ibc-go if the given Acknowledgement instance should be written to state or not.
```

Comment:
Technically the ack is always written to state when not nil. The `Success()` is about the callback data. You can see this best in the [code](https://github.com/cosmos/ibc-go/blob/main/modules/core/keeper/msg_server.go#L455-L472)

Reply:
What would be a better rephasing of this sentence?

Reply:
```suggestion
	// Success tells ibc-go if the state should be persisted. During the "OnRecvPacket" callback execution, all state is hold in a cache store and committed later. Note: events are always persisted.
```